README.md

```diff
@@ -26,134 +26,154 @@ composer require opentracing/opentracing
 ## Usage
 
 When consuming this library one really only need to worry about a couple of key
-abstractions: the `Tracer::startSpan` method, the `Span` interface, and binding
-a `Tracer` at bootstrap time. Here are code snippets demonstrating some important
-use cases:
+abstractions: the `Tracer::startActiveSpan` and `Tracer::startManualSpan` method,
+the `Span` interface, and binding a `Tracer` at bootstrap time. Here are code snippets
+demonstrating some important use cases:
 
 ### Singleton initialization
 
 The simplest starting point is to set the global tracer. As early as possible, do:
 
 ```php
-    use OpenTracing\GlobalTracer;
-    use AnOpenTracingImplementation\MyTracer;
-    
-    GlobalTracer::set(new MyTracer());
+use OpenTracing\GlobalTracer;
+
+GlobalTracer::set(new MyTracerImplementation());
 ```
 
 ### Creating a Span given an existing Request
 
-To start a new `Span`, you can use the `StartSpanFromContext` method.
+To start a new `Span`, you can use the `startActiveSpan` method.
 
 ```php
-    use OpenTracing\Formats;
-    use OpenTracing\GlobalTracer;
+use OpenTracing\Formats;
+use OpenTracing\GlobalTracer;
 
-    ...
+...
 
-    $spanContext = GlobalTracer::get()->extract(
-        Formats\HTTP_HEADERS,
-        $request->getHeaders()
-    );
+$spanContext = GlobalTracer::get()->extract(
+    Formats\HTTP_HEADERS,
+    getallheaders()
+);
+
+function doSomething(SpanContext $spanContext, ...) {
+    ...
     
-    function doSomething(SpanContext $spanContext, ...) {
-        ...
-        
-        $span = GlobalTracer::get()->startSpan('my_span', [
-        	'child_of' => $spanContext,
-        ]);
-        
-        ...
-        
-        $span->log([
-            'event' => 'soft error',
-            'type' => 'cache timeout',
-            'waiter.millis' => 1500,
-        ])
-        
-        $span->finish();
-    }
+    $span = GlobalTracer::get()->startManualSpan('my_span', ['child_of' => $spanContext]);
+    
+    ...
+    
+    $span->log([
+        'event' => 'soft error',
+        'type' => 'cache timeout',
+        'waiter.millis' => 1500,
+    ])
+    
+    $span->finish();
+}
 ```
 
 ### Starting an empty trace by creating a "root span"
 
-It's always possible to create a "root" `Span` with no parent or other causal
-reference.
+It's always possible to create a "root" `Span` with no parent or other causal reference.
 
 ```php
-    $span = $tracer->startSpan('my_span');
-    ...
-    $span->finish();
+$span = $tracer->startActiveSpan('my_first_span');
+...
+$span->finish();
 ```
 
-### Creating a (child) Span given an existing (parent) Span
+#### Creating a child span assigning parent manually
 
 ```php
-    function xyz(Span $parentSpan, ...) {
-        ...
-        $span = GlobalTracer::get()->startSpan(
-        	'my_span',
-        	[
-        		'child_of' => $parentSpan,
-        	]
-        );
-        
-        $span->finish();
-        ...
-    }
+$parent = GlobalTracer::get()->startManualSpan('parent');
```

Comment:
in Java 0.31 the "manual" functions have been deprecated. Is startSpan/startActiveSpan pair not good?

Reply:
I would highly suggest to keep an API around to create manual spans as a fallback for event loops that don't have the needed hooks to update the GlobalTracer, e.g. https://github.com/sabre-io/event

Reply:
Although I find `startActiveSpan` to be clearer. What do other languages use for naming?

Reply:
As far as I see, Java uses `startSpan/startActiveSpan` and Python will probably use the same. Ruby does not even implement this.